### ISSUE TITLE
AoB scaning always results in System.AggregateException

### ISSUE THREAD

**Issue by Nik2022** [erfg12/memory.dll#23]
I'm just trying to search an array of bytes. It always results in a System.AggregateException. Reading and writing inside the hooked applications memory works fine. Searching the AOB manually with Cheat Engine get the estimated result.

Here is my code:
 `long x = memory.AoBScan("base", memory.procs.MainModule.ModuleMemorySize, "74 1D 8A 54").Result;`
Also tried to run this as an new thread and using the await keyword.
The wikisite for aobscanning seems  to be outdated. It's using the old MemLib function.

I'm using NetFramework 4.5 and Memory.dll v1.2.15 from NuGet Packages. My application runs with admin privileges.

Here is the full exception;
System.AggregateException HResult=0x80131500 Message=at least one error occurred Source=mscorlib StackTrace: at System.Threading.Tasks.Task.ThrowIfExceptional(Boolean includeTaskCanceledExceptions) at System.Threading.Tasks.Task`1.GetResultCore(Boolean waitCompletionNotification) at MoneyHack.Program.findAddress() in J:\csharp\MoneyHack

What i'm doing wrong?

**Comment by hollow87** [comment]
`AggregateException` is an exception happening in the task it is wrapped use the `await` keyword and it should stop being an `AggregateException` and will be a different one.  If you need it to run synchronously use `.GetAwaiter().GetResult()`

**Comment by Nik2022** [comment]
Ok thank you for your quick respond. Sorry for this stupid  mistake. I'm new to C#. I added this to my code and it look like this now:
`long x = memory.AoBScan("base+0x00", memory.procs.MainModule.ModuleMemorySize, "74 1D 8A 54").GetAwaiter().GetResult();` But it always return 0. What did i wrong this time? I just want to search inside the entire memory of the hooked application for the byte array. I also checked two times that the byte array exits with cheat engine. 
Also thank you for your efforts
